components/feature/addons/src/main/java/mozilla/components/feature/addons/Addon.kt

```diff
@@ -80,11 +79,40 @@ data class Addon(
 
     /**
      * Returns a list of id resources per each item on the [Addon.permissions] list.
+     * Holds the state of the installed web extension of this add-on.
+     *
+     * @property id The ID of the installed web extension.
+     * @property version The installed version.
+     * @property enabled Indicates if this [Addon] is enabled to interact with web content or not,
+     * defaults to false.
+     * @property optionsPageUrl the URL of the page displaying the
+     * options page (options_ui in the extension's manifest).
+     */
+    @Parcelize
+    data class InstalledState(
+        val id: String,
+        val version: String,
+        val optionsPageUrl: String,
+        val enabled: Boolean = false
+    ) : Parcelable
+
+    /**
+     * Returns a list of id resources per each item on the [AddOn.permissions] list.
      */
     fun translatePermissions(): List<Int> {
-        return permissions.mapNotNull { it -> permissionToTranslation[it] }
+        return permissions.mapNotNull { permissionToTranslation[it] }
     }
 
+    /**
+     * Returns whether or not this [Addon] is currently installed.
+     */
+    fun isInstalled() = installedState != null
```

Comment:
nit: 
```    
val isInstalled: Boolean
    get() = installedState != null
```

Reply:
hm why do you think it's better to turn this into a property? We don't allow mutating the val anyway.

Reply:
Hm now that I think about it, `sInstalled` would never be mutable so I guess func makes more sense